src/ui/widget-forms/publish-single-content-config/SingleContentConfigFormBody.js

```diff
@@ -28,18 +30,41 @@ export class SingleContentConfigFormBody extends PureComponent {
   componentDidMount() {
     const { onDidMount } = this.props;
     onDidMount();
+
+    // fetch content from URL params
+    const queryString = window.location.search;
+    if (queryString.includes('contentId')) {
+      const urlParams = new URLSearchParams(queryString);
+      const contentId = urlParams.get('contentId');
+      this.fetchContentById(contentId);
+    }
   }
 
   componentDidUpdate(prevProps) {
     const { chosenContent } = this.props;
     const { selectedContent } = this.state;
+    const isNewContent = window.location.search.includes('contentId');
     if (
       prevProps.chosenContent !== chosenContent
       && !isEmpty(chosenContent)
       && isEmpty(selectedContent)
+      && !isNewContent
     ) {
-      // eslint-disable-next-line react/no-did-update-set-state
-      this.setState({ selectedContent: chosenContent });
+      if (chosenContent.status) {
+        // eslint-disable-next-line react/no-did-update-set-state
```

Comment:
sure about this one?

Reply:
`chosenContent` usually has all attributes of a content, including its `status`, but when the config form is called after a new content is created, the callback function only passes the `contentId`, in that case, I need to fetch the remaining data of the content to show it in the new table.